eva-lib/src/main/java/uk/ac/ebi/eva/lib/repository/VariantEntityRepositoryImpl.java

```diff
@@ -111,12 +124,101 @@ public Long countByRegionsAndComplexFilters(List<Region> regions, List<VariantEn
             exclude.forEach(e -> query.fields().exclude(e));
         }
 
-        return mongoTemplate.find(query, VariantEntity.class);
+        List<VariantEntity> variantEntities = mongoTemplate.find(query, VariantEntity.class);
+
+        Table<String, String, Map<String, String>> studyFileIdsToPositionSamples = studyFileIdsToPositionSamples();
+
+        variantEntities = updateVariantEntitiesSampleNames(variantEntities, studyFileIdsToPositionSamples);
+
+        return variantEntities;
+    }
+
+    private Table<String, String, Map<String, String>> studyFileIdsToPositionSamples() {
+        List<VariantSourceEntity> variantSourceEntities = variantSourceEntityRepository.findAll();
 
+        Table<String, String, Map<String, String>> studyFileIdsToPositionSamples = HashBasedTable.create();
+
+        for (VariantSourceEntity variantSourceEntity : variantSourceEntities) {
+            if (variantSourceEntity.getSamplesPosition() == null) {
+                continue;
+            }
+
+            String studyId = variantSourceEntity.getStudyId();
+            String fileId = variantSourceEntity.getFileId();
+
+            Map<String, Integer> samplesPosition = variantSourceEntity.getSamplesPosition();
+            Map<String, String> positionSamples = new HashMap<>();
+            for(Map.Entry<String, Integer> entry : samplesPosition.entrySet()) {
+                positionSamples.put(Integer.toString(entry.getValue()), entry.getKey());
+            }
+
+            studyFileIdsToPositionSamples.put(studyId, fileId, positionSamples);
+        }
+
+        return studyFileIdsToPositionSamples;
+    }
+
+    private List<VariantEntity> updateVariantEntitiesSampleNames(List<VariantEntity> variantEntities,
+                                                                 Table<String, String, Map<String, String>>
+                                                                         studyFileIdsToPositionSamples) {
+
+        return variantEntities.stream().map(
+                variantEntity -> updateVariantSourceEntrySampleNames(variantEntity, studyFileIdsToPositionSamples)
+        ).collect(Collectors.toList());
+    }
+
+    private VariantEntity updateVariantSourceEntrySampleNames(VariantEntity variantEntity,
```

Comment:
For some reason this happened when I used intellij refactoring- it changed both method names.